### PR TITLE
241-Custom-Emoji-Tests

### PR DESCRIPTION
**Pull request by KyleNewton10**
This PR adds testing to the CustomEmoji package

### CODE REVIEW

File: lombok.config [hunk 1]
```diff
@@ -0,0 +1 @@
+lombok.addLombokGeneratedAnnotation = true
```
Comment (KyleNewton10) [inline lombok.config:1]: This file excludes any generated Lombok code from the previously implemented Jacoco Test Report

This removes the need to test getter/setter autogenerated code

File: src/main/java/com/sublinks/sublinksapi/customemoji/services/CustomEmojiService.java [hunk 2]
```diff
@@ -32,12 +32,14 @@ public CustomEmoji updateCustomEmoji(final CustomEmoji customEmoji, final Iterab
 
   private CustomEmoji addKeywords(final CustomEmoji customEmoji, final Iterable<String> keywords) {
     var customEmojiKeywords = new ArrayList<CustomEmojiKeyword>();
-    for (var keyword : keywords) {
-      customEmojiKeywords.add(
-          CustomEmojiKeyword.builder()
-              .keyword(keyword.toLowerCase().trim())
-              .emoji(customEmoji)
-              .build());
+    if (keywords != null) {
```
Comment (KyleNewton10) [inline src/main/java/com/sublinks/sublinksapi/customemoji/services/CustomEmojiService.java:35]: Found a bug here that would throw an NPE if a null list was passed through to the update method. I've updated this to check for null and gracefully return an empty list.
Alternatively if there's a need to always have a list of keywords associated we can throw an exception and fail the update if the iterator is null

File: src/test/java/com/sublinks/sublinksapi/customemoji/dto/CustomEmojiKeywordTest.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class CustomEmojiKeywordTest {
+
+  @Test
+  public void testEquals() {
```
Comment (KyleNewton10) [inline src/test/java/com/sublinks/sublinksapi/customemoji/dto/CustomEmojiKeywordTest.java:15]: Regarding Test Names - Is there an established convention that has been set for naming? I see a couple different methods implemented so far

File: src/test/java/com/sublinks/sublinksapi/customemoji/services/CustomEmojiServiceTest.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (jgrim) [inline src/test/java/com/sublinks/sublinksapi/customemoji/services/CustomEmojiServiceTest.java:21]: Please review the style guide: https://sublinks.org/docs/style_guide.html

Wildcard imports are not allowed.

Comment (KyleNewton10) [reply src/test/java/com/sublinks/sublinksapi/customemoji/services/CustomEmojiServiceTest.java]: Sorry about that miss, I've reviewed all the code for styleguide violations, I believe I've covered everything this time

**Review by jgrim — APPROVED** [review]

**Pull request merged**